internal/gcs/process.go

```diff
@@ -218,6 +283,12 @@ func (p *Process) ResizeConsole(ctx context.Context, width, height uint16) (err
 		trace.StringAttribute("cid", p.cid),
 		trace.Int64Attribute("pid", int64(p.id)))
 
+	p.gcLock.RLock()
+	defer p.gcLock.RUnlock()
+	if p.gc == nil {
+		return hcs.ErrAlreadyClosed
+	}
```

Comment:
This should use the helper function

Reply:
Since the helper function locks and unlocks gcLock for reading, I didnt want a code flow with repeated locks and unlocks of the same mutex:
```go
	// p.gcClosed()
	p.gcLock.RLock()
	if p.gc == nil {
		return hcs.ErrAlreadyClosed
	}
	p.gcLock.RUnlock()

	p.gcLock.RLock()
	defer p.gcLock.RUnlock()
	req := containerResizeConsole{...}
	return p.gc.brdg.RPC(ctx, rpcResizeConsole, &req, &resp, true)
```

Reply:
I can switch to that if you think its cleaner and not a consideration